plugins/dominant-color-images/class-dominant-color-image-editor-gd.php

```diff
@@ -32,11 +32,11 @@ public function get_dominant_color() {
 		}
 		// The logic here is resize the image to 1x1 pixel, then get the color of that pixel.
 		$shorted_image = imagecreatetruecolor( 1, 1 );
-		$image_width   = imagesx( $this->image );
-		$image_height  = imagesy( $this->image );
-		if ( false === $shorted_image || false === $image_width || false === $image_height ) {
```

Comment:
In PhpStorm's stubs, it defines:

```php
/**
 * Create a new true color image
 * @link https://php.net/manual/en/function.imagecreatetruecolor.php
 * @param int $width <p>
 * Image width.
 * </p>
 * @param int $height <p>
 * Image height.
 * </p>
 * @return resource|GdImage|false an image resource identifier on success, false on errors.
 */
#[Pure]
function imagecreatetruecolor(int $width, int $height): GdImage|false {}

/**
 * Get image width
 * @link https://php.net/manual/en/function.imagesx.php
 * @param resource|GdImage $image
 * @return int|false Return the width of the image or false on
 * errors.
 */
#[Pure]
function imagesx(GdImage $image): int {}

/**
 * Get image height
 * @link https://php.net/manual/en/function.imagesy.php
 * @param resource|GdImage $image
 * @return int|false Return the height of the image or false on
 * errors.
 */
#[Pure]
function imagesy(GdImage $image): int {}
```

All three have phpdoc that say they can return `false`, but only `imagecreatetruecolor()` has the PHP type hint which says it can return `false`, and this is confirmed in the PHP docs: https://www.php.net/manual/en/function.imagecreatetruecolor.php

In contrast, the PHP docs for the other functions say nothing about them returning false:

https://www.php.net/manual/en/function.imagesx.php
https://www.php.net/manual/en/function.imagesy.php

So it seems PhpStorm's docs are incorrect.